File: app/src/main/java/com/nicoqueijo/android/currencyconverter/Conversion/CurrencyConversion.java
package com.nicoqueijo.android.currencyconverter.Conversion;

public class CurrencyConversion {

    private static final String TAG = "CurrencyConversion";

    public static double currencyConverter(int amount, double fromRate, double toRate) {
        double valueInDollars = convertAnyCurrencyToDollar(amount, fromRate);
        double valueInTargetCurrency = convertDollarToAnyCurrency(valueInDollars, toRate);
        return valueInTargetCurrency;
    }

    private static double convertAnyCurrencyToDollar(int amount, double fromRate) {
        return (amount / fromRate);
    }

    private static double convertDollarToAnyCurrency(double dollarValue, double toRate) {
        return (dollarValue * toRate);
    }
}

File: app/src/main/java/com/nicoqueijo/android/currencyconverter/kotlin/util/CurrencyConversion.kt
package com.nicoqueijo.android.currencyconverter.kotlin.util

import java.math.BigDecimal
import java.math.RoundingMode

object CurrencyConversion {

    fun convertCurrency(amount: BigDecimal, fromRate: Double, toRate: Double): BigDecimal {
        val valueInDollars = convertAnyCurrencyToDollar(amount, fromRate)
        return convertDollarToAnyCurrency(valueInDollars, toRate)
    }

    private fun convertAnyCurrencyToDollar(amount: BigDecimal, fromRate: Double): BigDecimal {
        val scale = 50
        return amount.divide(BigDecimal.valueOf(fromRate), scale, RoundingMode.HALF_UP)
    }

    private fun convertDollarToAnyCurrency(dollarValue: BigDecimal, toRate: Double): BigDecimal {
        return dollarValue.multiply(BigDecimal.valueOf(toRate))
    }
}